File: LeetCode/Easy/605/605.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        if n == 0:
            return True
        count = 0
        for i in range(len(flowerbed)):
            left = (i == 0) or (flowerbed[i-1] == 0)
            right = (i == len(flowerbed) -1) or (flowerbed[i+1] ==0)
            if(flowerbed[i] == 0 and left and right):
                flowerbed[i] = 1
                count += 1

                if(count == n):
                    return True
        else:
            return False

File: LeetCode/605/605.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int count = 0;
        bool left, right;
        if(n==0){
            return true;
        }
        for(int i =0; i< flowerbed.size(); i++){
            left = (i == 0) || (flowerbed[i-1]==0);
            right = (i == flowerbed.size() - 1) || (flowerbed[i+1] == 0);
            if(flowerbed[i]==0 && left && right){
                flowerbed[i] = 1;
                count++;
            }
            if(count == n){
                return true;
            }
        }
        return false;
    }
};